### PR TITLE
Adds attributes to token and translations to template

### PR DESCRIPTION
**Pull request by arothuis**
[Ticket 106955902](https://www.pivotaltracker.com/story/show/106955902) and [ticket 106956276](https://www.pivotaltracker.com/story/show/106956276)

Done:
* Attributes to token
* Translation keys in template
* Translations

### CODE REVIEW

**Comment by arothuis** [comment]
@rjkip

File: src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php [hunk 1]
```diff
@@ -18,27 +18,40 @@
 
 namespace OpenConext\ProfileBundle\Security\Authentication\Entity;
 
-class User
+use Surfnet\SamlBundle\SAML2\Attribute\AttributeSet;
+
+class User implements \Serializable
```
Comment (rjkip) [inline src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php]: Please import `\Serializable`

File: src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php [hunk 2]
```diff
@@ -49,11 +62,6 @@ class User
      */
     public function __toString()
     {
-        return serialize([
-            $this->nameId,
-            $this->institution,
-            $this->email,
-            $this->commonName
-        ]);
+        return serialize($this);
```
Comment (rjkip) [inline src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php]: `User::__toString()` is used for:

0. change detection in `AbstractToken::setUser()`;
0. and display in Symfony's web toolbar.

Perhaps it is good to change this to `'User('.sha1(serialize($this)).')'`. Change detection will still work and the web toolbar doesn't display a logged in user that wraps and covers the entire width of the screen. Or perhaps get the EPTI from the attributes. Opinions, @DRvanR?

Comment (DRvanR) [inline src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php]: Y U NO

File: src/OpenConext/ProfileBundle/Resources/views/Profile/overview.html.twig [hunk 3]
```diff
@@ -11,19 +11,27 @@
...
             <thead>
             <tr>
-                <th>Attribute</th>
-                <th>Value</th>
+                <th>{{ 'profile.table.attribute_name'|trans }}</th>
+                <th>{{ 'profile.table.attribute_value'|trans }}</th>
             </tr>
             </thead>
             <tbody>
+            {% for attribute in attributes %}
```
Comment (DRvanR) [inline src/OpenConext/ProfileBundle/Resources/views/Profile/overview.html.twig]: believe you could just use `app.user.attributes`, prevents having to pass the attributes as they are already available?

File: src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php (hunk 2 above)
Comment (arothuis) [inline src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php]: LOL

**Comment by arothuis** [comment]
@DRvanR, @rjkip

File: src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php [hunk 4]
```diff
@@ -18,27 +18,56 @@
...
-    public $institution;
+    private $attributes;
 
     /**
-     * @var string
+     * @param string $nameId
+     * @param AttributeSet $attributes
      */
-    public $email;
+    public function __construct($nameId, AttributeSet $attributes)
```
Comment (DRvanR) [inline src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php]: As discussed, please make the class final with a named constructor

File: src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php [hunk 5]
```diff
@@ -18,27 +18,56 @@
...
-     * @var string
+     * @return AttributeSet
      */
-    public $commonName;
+    public function getAttributes()
+    {
+        return $this->attributes;
+    }
+
+    public function serialize()
```
Comment (DRvanR) [inline src/OpenConext/ProfileBundle/Security/Authentication/Entity/User.php]: as discussed, serialization methods can be removed.

**Comment by DRvanR** [comment]
Also ensure you've merged in latest master and resolved BC breaks (Assert has moved namespace) as well as that you've updated the StepupSamlBundle to `^1.4`

**Comment by arothuis** [comment]
@DRvanR, @rjkip

**Comment by rjkip** [comment]
:guitar:

**Comment by DRvanR** [comment]
:+1:

**Pull request merged**